internal/app.go

```diff
@@ -0,0 +1,187 @@
+package internal
+
+import (
+	"embed"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	ics "github.com/arran4/golang-ical"
+	"github.com/gin-gonic/gin"
+	"io"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+const src = "https://campus.tum.de/tumonlinej/ws/termin/ical?pStud=%s&pToken=%s"
```

Comment:
Consider code simplicity by putting strings only used once not at a file global level, but scope them closely to where they are actually used.